### PR TITLE
[AMDGPU][MIR] Serialize SpillPhysVGPRs

### PR DESCRIPTION
**Pull request by optimisan**

### CODE REVIEW

File: llvm/test/CodeGen/MIR/AMDGPU/spill-phys-vgprs.mir [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    S_NOP 0
+  bb.1:
+    $sgpr40 = S_MOV_B32 0
+    $sgpr41 = S_MOV_B32 1
+
+...
+
+# CHECK-LABEL: name: parse_none
+# CHECK: machineFunctionInfo:
+# CHECK-NOT: spillPhysVGPRs:
```
Comment (arsenm) [inline llvm/test/CodeGen/MIR/AMDGPU/spill-phys-vgprs.mir:20]:
```suggestion
# CHECK-NOT: spillPhysVGPRs
```

-NEXT checks would be more reliable

File: llvm/lib/Target/AMDGPU/AMDGPUTargetMachine.cpp [hunk 2]
```diff
@@ -1718,6 +1718,13 @@ bool GCNTargetMachine::parseMachineFunctionInfo(
     MFI->reserveWWMRegister(ParsedReg);
   }
 
+  for (const auto &YamlRegStr : YamlMFI.SpillPhysVGPRS) {
+    Register ParsedReg;
+    if (parseRegister(YamlRegStr, ParsedReg))
+      return true;
```
Comment (arsenm) [inline llvm/lib/Target/AMDGPU/AMDGPUTargetMachine.cpp:1724]: Test the failing parse case

File: llvm/lib/Target/AMDGPU/SIMachineFunctionInfo.h [hunk 3]
```diff
@@ -275,6 +275,7 @@ struct SIMachineFunctionInfo final : public yaml::MachineFunctionInfo {
   // TODO: 10 may be a better default since it's the maximum.
   unsigned Occupancy = 0;
 
+  SmallVector<StringValue, 2> SpillPhysVGPRS;
```
Comment (arsenm) [inline llvm/lib/Target/AMDGPU/SIMachineFunctionInfo.h:278]: I see the name was copied from the existing MFI field, but this is not a good name. We ought to rename this to reflect the special usage; this isn't just any spilled physical vgprs

File: llvm/test/CodeGen/MIR/AMDGPU/spill-phys-vgprs.mir [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+name: parse_one
+machineFunctionInfo:
+  spillPhysVGPRs: ['$vgpr0']
+body: |
+  bb.0:
+    S_ENDPGM 0
+
+...
+
+# CHECK-LABEL: name: parse_one
```
Comment (arsenm) [inline llvm/test/CodeGen/MIR/AMDGPU/spill-phys-vgprs.mir:45]: Test a case with multiple

File: llvm/lib/Target/AMDGPU/SIMachineFunctionInfo.h (hunk 3 above)
Comment (arsenm) [reply llvm/lib/Target/AMDGPU/SIMachineFunctionInfo.h:278]: WWMPhysVGPRs?

File: llvm/test/CodeGen/MIR/AMDGPU/spill-phys-vgprs-invalid.mir [hunk 5]
```diff
@@ -0,0 +1,12 @@
+# RUN: not llc -mtriple=amdgcn-amd-amdhsa -run-pass=none -o /dev/null %s 2>&1 | FileCheck %s --check-prefix=ERR
+
+---
+name: invalid_reg_spill_phys_vgprs
+machineFunctionInfo:
+# ERR: [[@LINE+1]]:21: unknown register name 'notareg'
+  spillPhysVGPRs: ['$notareg']
+body: |
```
Comment (arsenm) [inline llvm/test/CodeGen/MIR/AMDGPU/spill-phys-vgprs-invalid.mir:8]: Check empty string and not a string case

**Review by arsenm — APPROVED** [review]
lgtm with nit

File: llvm/lib/Target/AMDGPU/SIMachineFunctionInfo.h (hunk 3 above)
Comment (optimisan) [reply llvm/lib/Target/AMDGPU/SIMachineFunctionInfo.h:278]: Okay

**Pull request merged**